benchmarks/persistence/persistence.py

```diff
@@ -0,0 +1,229 @@
+"""Classes for persisting benchmark results."""
+
+import json
+import os
+from typing import Protocol
+
+import boto3
+import boto3.session
+from attr import define, field
+from attrs.validators import instance_of
+from git import Repo
+
+from benchmarks import Benchmark, Result
+
+
+class PersistenceObjectInterface(Protocol):
+    """Interface for constructing the path of a result object."""
+
+    def get_full_file_path(self) -> str:
+        """Construct the path of a result object.
+
+        Returns:
+            The path of the result object.
+        """
+
+    def get_object(self) -> dict:
```

Comment:
I wonder why this needs to be part of the interface definition. If you do that, it means that different backends can potentially store the same object in different formats, which seems weird to me. My perspective is that, as long as the object is correctly serialized, the obtained representation should be compatible with all backends, i.e. completely agnostic to where it will be persisted

Reply:
I thought you wanted a separation between where to store data and how to store it. Since we have divided the data into results and benchmarks, the construction of the object to store is not quite clear anymore. I think there are two options:

1. The current structure, where a class is responsible for defining the object and another for storing it.
2. Alternatively, the path could become part of the object to store again, and the function would only receive a dictionary. If the construction occurs within `__main__`, there would be no need for a separate path class at all.

Reply:
I've changed it now but since there are information necessary to build the the right key, the object constructing has to have knowledge about the benchmark and result objects.